src/celengine/body.h

```diff
@@ -96,76 +97,79 @@ class RingSystem
 };
 
 
+// Object class enumeration:
+// All of these values must be powers of two so that they can
+// be used in an object type bit mask.
+//
+// The values of object class enumerants cannot be modified
+// without consequence. The object orbit mask is a stored user
+// setting, so there will be unexpected results when the user
+// upgrades if the orbit mask values mean something different
+// in the new version.
+//
+// * Planet, Moon, Asteroid, DwarfPlanet, and MinorMoon all behave
+// essentially the same. They're distinguished from each other for
+// user convenience, so that it's possible to assign them different
+// orbit and label colors, and to categorize them in the solar
+// system browser.
+//
+// * Comet is identical to the asteroid class except that comets may
+// be rendered with dust and ion tails.
+//
+// Other classes have different default settings for the properties
+// Clickable, VisibleAsPoint, Visible, and SecondaryIlluminator. These
+// defaults are assigned in the ssc file parser and may be overridden
+// for a particular body.
+//
+// * Invisible is used for barycenters and other reference points.
+// An invisible object is not clickable, visibleAsPoint, visible, or
+// a secondary illuminator.
+//
+// * SurfaceFeature is meant to be used for buildings and landscape.
+// SurfaceFeatures is clickable and visible, but not visibleAsPoint or
+// a secondary illuminator.
+//
+// * Component should be used for parts of spacecraft or buildings that
+// are separate ssc objects. A component is clickable and visible, but
+// not visibleAsPoint or a secondary illuminator.
+//
+// * Diffuse is used for gas clouds, dust plumes, and the like. They are
+// visible, but other properties are false by default. It is expected
+// that an observer will move through a diffuse object, so there's no
+// need for any sort of collision detection to be applied.
+//
+// * Stellar is a pseudo-class used only for orbit rendering.
+//
+// * Barycenter and SmallBody are not used currently. Invisible is used
+// instead of barycenter.
+enum class BodyClassification : std::uint16_t
```

Comment:
why limiting to `std::uint16_t`?

Reply:
ah right, had done this on top of the other PR with optional features but split it out... made sense in the context of the layout there but probably going to do some changes there as well